### PR TITLE
adding hex to dec function 

### PR DESCRIPTION
**Pull request by aylapear**
- adding new function `hex_to_dec_code()` 
- purpose is to help us convert PIT tag hexadecimal codes into the corresponding decimal codes
- styler ran
- lint ran
- all tests and R CMD check passing 
- adding new function mention with fledge notation
- tests written for function

### CODE REVIEW

**Comment by aylapear** [comment]
Hey @nadinehussein, let me know if this suits your needs or if you want any more customization on it. And then we can have Joe review it before approval

**Review by nadinehussein — COMMENTED** [review]
paste 900 on front of 12 digit hex codes after conversion and if 11 digits pad with 0 at front

**Comment by aylapear** [comment]
@nadinehussein I added a second function called `hex_to_dec_pit_tag()` 
- this wrapper function converts the hex to dec and 
- it checks the length of the dec code
  - if 11 or under it pads zeros to the left to make it 12 digits long
  - if 12 digits long do nothing
  - if 13 or more digits it returns `NA` 
- by default it appends 900 to the front of the code once it is 12 digits long

**Comment by aylapear** [comment]
update so it pads 11 but returns NA if less then 11 digits returned

**Comment by aylapear** [comment]
Function names change to `hex_to_dec()` and `ps_h2d_tag()`

**Comment by nadinehussein** [comment]
looks good to me get Joe to review?

**Comment by joethorley** [comment]
Looks good but several functions seem to be defined twice - this obviously needs fixing.

**Comment by joethorley** [comment]
This code
```
dec[is.nan(dec)] <- NA
```

should be 
```
dec[is.nan(dec)] <- NA_character_
```

for type stability in case all values are nan. The former might produce a logical instead of character vector.

**Comment by joethorley** [comment]
You should also add some tests on zero length vectors ie character() and integer() to ensure it can handle these edge cases. ie should return zero length vectors of the correct type.

**Comment by aylapear** [comment]
- removed duplicate function definitions 
- updated NA assignment to be `NA_real_` to ensure type is known
- added zero length vector test 
   - `character()` passes and outputs a double as expected
   - `numeric()` fails as the function requires a string to be passed based on the chk at the start of the function.

**Comment by aylapear** [comment]
@joethorley Ready for another review

File: R/hex-to-dec.R
Comment (joethorley) [inline R/hex-to-dec.R:24]: should it be returning integers (and using NA_integer_) or is this not possible with the numbers because they are so big?

**Review by joethorley — APPROVED** [review]
Only 1 pending comment

File: R/hex-to-dec.R
Comment (aylapear) [reply R/hex-to-dec.R:24]: You got it, the numbers are too big. When you set them to be integer it turns them to be NA

Comment (aylapear) [reply R/hex-to-dec.R:24]: I thought real was slightly better than character but not sure what the best is in a case like this.

Comment (joethorley) [reply R/hex-to-dec.R:24]: I'm not sure either - is there a chance of two pit tags looking the same but not matching because of a difference at a small number of decimal places?

Comment (aylapear) [reply R/hex-to-dec.R:24]: I haven't seen one convert to have a decimal number, the ones I have seen always convert to whole numbers. 

My understanding of a hex code is it will always convert to a whole number/integer, the issue we have is that R can't seem to handle these large integers. Each hex coding should always generate the same whole number.

If it appends a decimal in the background because the value in a binary representation actually has decimal points maybe this would cause an issue. I am not sure nor have I found an example that does that. 

I have tested the function against the set of values we have from one study and they all matched up perfectly in that case. 

If you feel character is `NA_character_` would be safer I can update to that.

Comment (joethorley) [reply R/hex-to-dec.R:24]: No I think we go with real and change to character if we encounter problems.

**Comment by aylapear** [comment]
@joethorley is this ready to be merged now? I am not sure based on the comment above vs the last comment sent to me regarding this in personal communications.

**Comment by joethorley** [comment]
Sorry - very confusing. I approved and then encountered a dataset with mixed PIT tag codes all as character. Once you update so it returns class character its ready to good.

**Comment by aylapear** [comment]
- Updated `hex_to_dec` function to output character vectors instead of numerical vectors
- Updated rules in `ps_h2d_tag()` function so any decimal values less then 12 digits are padded with zeros on the left to ensure the decimal value is 12 digits long before the country code is appended to the decimal value to make it a valid pit tag code

**Comment by joethorley** [comment]
Well done on getting it over the line!

**Pull request merged**
